components/p3a_utils/feature_usage.cc

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/p3a_utils/feature_usage.h"
+
+#include "base/metrics/histogram_functions.h"
+#include "brave/components/p3a_utils/bucket.h"
+#include "brave/components/time_period_storage/monthly_storage.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+
+namespace p3a_utils {
+
+void RegisterFeatureUsagePrefs(PrefRegistrySimple* registry,
+                               const char* first_use_time_pref_name,
+                               const char* last_use_time_pref_name,
+                               const char* used_second_day_pref_name,
+                               const char* days_in_month_used_pref_name) {
+  if (first_use_time_pref_name) {
+    registry->RegisterTimePref(first_use_time_pref_name, base::Time());
+  }
+  if (last_use_time_pref_name) {
+    registry->RegisterTimePref(last_use_time_pref_name, base::Time());
+  }
+  if (used_second_day_pref_name) {
```

Comment:
What's the advantage to passing `nullptr` for these pref names? How do we prevent callers from `RecordFeatureNewUserReturning()` when they haven't registered the appropriate prefs? Maybe a DCHECK would be better here?

Could also catch typos by `static_assert`ing that the prefs are distinct.

Reply:
A particular feature may not need to register all the prefs, depending on which metrics are being recorded from `feature_usage`. line 78 in `components/brave_wallet/browser/brave_wallet_prefs.cc` is an example: it does not register the "days in month" pref because it doesn't record the "days in month" metric. But now that I'm thinking about it, we can add `DCHECK`s for the first & last use timestamps (since they're always mandatory), and keep the last two prefs as optional.